lib/puppet/pops/types/type_calculator.rb

```diff
@@ -875,64 +618,50 @@ def infer_Symbol(o)
 
   # @api private
   def infer_TrueClass(o)
-    Types::PBooleanType.new()
+    Types::PBooleanType::DEFAULT
```

Comment:
Note: we should consider making boolean parameterized. (We should log a ticket for that). Reason being you cannot precisely compare booleans with the type system and instead have to use a test on value.